### DISCUSSION TITLE
Can we regex search in filters?

### DISCUSSION THREAD

**Discussion by glitchedgitz** [pocketbase/pocketbase#3120] [Q&A]
I didn't find a regex search in the docs.
If not possible, If there any alternative way to achieve this?

**Comment by ganigeorgiev** [comment]
There is no regex support for the filters, only wildcard matching with the `~` operator.

> If there any alternative way to achieve this?

Only with a custom route and manually querying the database. Note that performing client-side regex searches with untrusted user provided patterns could have significant impact on the performance of your application and I'd generally advice against it, unless this is not a concern for your specific use case.

**Reply by glitchedgitz** [reply]
Yes, I am aware of ddos attacks, but mine is a desktop application where this is a crucial feature.

What is the best approach to achieve regex search with an operator in filter like `~~` so I can use your original awesome API and SDK?

**Reply by ganigeorgiev** [reply]
There is no easy answer to this unfortunately as the filters syntax cannot be extended at the moment. 

Actually, I'm not even sure whether the SQLite `REGEXP` expression can be used since we don't specify explicitly a `regexp()` function and probably will return an error or result in `false` evaluation (_and the behavior and how this function is registered will differ depending whether the CGO or pure Go port is used_).
